test/CoreTest.cpp

```diff
@@ -862,6 +868,92 @@ TEST_CASE( "no_file_ending", "[core]" )
                         Catch::Equals("Unknown file format! Did you specify a file ending?"));
     REQUIRE_THROWS_WITH(Series("./new_openpmd_output_%05T", Access::CREATE),
                         Catch::Equals("Unknown file format! Did you specify a file ending?"));
+    {
+        Series(
+            "../samples/no_extension_specified",
+            Access::CREATE,
+            "{\"backend\": \"json\"}" );
+    }
+    REQUIRE(
+        auxiliary::file_exists( "../samples/no_extension_specified.json" ) );
+}
+
+TEST_CASE( "backend_via_json", "[core]" )
+{
+    std::string encodingVariableBased =
+        "{\"backend\": \"json\", \"iteration_encoding\": \"variable_based\"}";
+    {
+        Series series(
+            "../samples/optionsViaJson",
+            Access::CREATE,
+            encodingVariableBased );
+        REQUIRE( series.backend() == "JSON" );
+        REQUIRE(
+            series.iterationEncoding() == IterationEncoding::variableBased );
+    }
+#if openPMD_HAVE_ADIOS2
+    {
+        Series series(
+            "../samples/optionsViaJson.bp",
+            Access::CREATE,
+            encodingVariableBased );
+        REQUIRE( series.backend() == "JSON" );
+        REQUIRE(
+            series.iterationEncoding() == IterationEncoding::variableBased );
+    }
+
+    {
+        Series series(
+            "../samples/optionsViaJsonOverwritesAutomaticDetection.sst",
+            Access::CREATE,
+            "{\"adios2\": {\"engine\": {\"type\": \"bp4\"}}}" );
+    }
+    REQUIRE( auxiliary::directory_exists(
+        "../samples/optionsViaJsonOverwritesAutomaticDetection.bp" ) );
+
+#if openPMD_HAVE_ADIOS1
+    setenv( "OPENPMD_BP_BACKEND", "ADIOS1", 1 );
```

Comment:
We currently handle this by having a CI test that sets the variable (and runs all BP tests with ADIOS1 then that support it).
Maybe that's simpler to keep so?

Reply:
The purpose of this test is to make it explicit that the JSON option `"backend": "adios2"` should have priority over the environment variable. Sure if we removed the `setenv()` this would still be covered by CI, but having the line in the test makes it explicit what this is about.

Reply:
> The purpose of this test is to make it explicit that the JSON option `"backend": "adios2"` should have priority over the environment variable.

Got it, makes sense to me. Thanks!

Reply:
(Maybe add this as a comment in it? :) )

Reply:
added :)